docs/components/api-item-member.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import createFormatters from 'documentation/lib/output/util/formatters';
+import LinkerStack from 'documentation/lib/output/util/linker_stack';
+import docs from '../components/api.json'; // eslint-disable-line import/no-unresolved
+import ApiItem from '../components/api-item';
+import Icon from '@mapbox/mr-ui/icon';
+
+const linkerStack = new LinkerStack({})
+    .namespaceResolver(docs, (namespace) => {
+        const slugger = new GithubSlugger();
+        return `#${slugger.slug(namespace)}`;
+    });
+
+const formatters = createFormatters(linkerStack.link);
+
+class ApiItemMember extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {disclosed: false};
+        this.hashChange = this.hashChange.bind(this);
+    }
+
+    href = m => {
+        return `#${m.namespace.toLowerCase()}`;
+    }
+
+    render() {
+        const member = this.props;
+        return (
+            <div>
+                <div>
+                    <div className='pt60 border-b border--gray-light' style={{ marginTop: '-40px' }} id={member.namespace.toLowerCase()} />
+                    <div
+                        className='clearfix cursor-pointer toggle-sibling pt18'
+                        onClick={(e) => {
+                            this.setState({disclosed: !this.state.disclosed});
+                            window.location = this.href(member)                    
+                            e.preventDefault();
+                        }}
+                    >
+                        <span className='txt-code truncate'>{member.name}</span>
+                        {member.kind === 'function' &&
+                            <span className='color-gray txt-code mr12' dangerouslySetInnerHTML={{__html: formatters.parameters(member, true)}}/>}
+                        <Icon name={`${this.state.disclosed ? 'caret-down' : 'caret-right'}`} themeIcon="fr" inline={true} />
+                    </div>
+                </div>
+
+                {this.state.disclosed &&
+                    <div className="clearfix toggle-target">
+                        <ApiItem nested={true} {...member}/>
+                    </div>}
+            </div>
+        );
+    }
+
+    hashChange() {
+        if (window.location.hash === this.href(this.props)) {
```

Comment:
I think you also need to add an `else` statement to set `disclosed` to `false` when the user clicks on another sidebar api member (but remain open if a user clicks another heading in the main content area?)

Reply:
I'm not sure we need to enforce that only one item be open at a time. Is that what you're getting at here? 😬